aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java

```diff
@@ -528,19 +526,19 @@ private static MethodHandle adaptSignature(FunctionType type, MethodHandle handl
         var argTypes = type.params();
         var argHandlers = new MethodHandle[type.params().size()];
         for (int i = 0; i < argHandlers.length; i++) {
-            argHandlers[i] = unboxerHandle(argTypes.get(i));
+            argHandlers[i] = convertFromLongHandle(argTypes.get(i));
         }
         MethodHandle result = filterArguments(handle, 0, argHandlers);
-        result = result.asSpreader(Value[].class, argTypes.size());
+        result = result.asSpreader(long[].class, argTypes.size());
 
         if (type.returns().isEmpty()) {
-            return result.asType(result.type().changeReturnType(Value[].class));
+            return result.asType(result.type().changeReturnType(long[].class));
         }
         if (type.returns().size() > 1) {
             return result;
         }
-        result = filterReturnValue(result, boxerHandle(type.returns().get(0)));
-        return filterReturnValue(result, ValueWrapper.HANDLE);
```

Comment:
I think `ValueWrapper` is unused now

Reply:
now it's a `LongArrayWrapper` to: `long -> long[]`